core/services/vrf/v2/listener_v2.go

```diff
@@ -441,15 +444,18 @@ func (lsn *listenerV2) processPendingVRFRequests(ctx context.Context) {
 				// to handle this specific situation, since we need to run the pipeline to get
 				// the VRF proof, abi-encode it, etc.
 				l.Warnw("Subscription not found - setting start balance to zero", "subID", subID, "err", err)
-				startBalance = big.NewInt(0)
+				startLinkBalance = big.NewInt(0)
 			} else {
 				// Most likely this is an RPC error, so we re-try later.
 				l.Errorw("Unable to read subscription balance", "err", err)
 				continue
 			}
 		} else {
 			// Happy path - sub is active.
-			startBalance = sub.Balance
+			startLinkBalance = sub.Balance()
+			if sub.VRFVersion == vrfcommon.V2Plus {
+				startEthBalance = sub.EthBalance()
```

Comment:
is this an extra RPC call? If so, I'd consider adding a joint getBalances function.

Reply:
Nope, this isn't an RPC call. The field will be available if we are on V2+, so it'll just be a Go struct field access.